evidence/reactor_test.go

```diff
@@ -404,55 +415,49 @@ func TestReactorBroadcastEvidence_Committed(t *testing.T) {
 	stateDB1 := initializeValidatorState(t, val, height)
 	stateDB2 := initializeValidatorState(t, val, height)
 
-	testSuites := createTestSuites(t, []sm.Store{stateDB1, stateDB2}, 0)
-	primary := testSuites[0]
-	secondary := testSuites[1]
+	rts := setup(t, []sm.Store{stateDB1, stateDB2}, 0)
+
+	primary := rts.nodes[0]
+	secondary := rts.nodes[1]
 
 	// add all evidence to the primary reactor
-	evList := createEvidenceList(t, primary.pool, val, numEvidence)
+	evList := createEvidenceList(t, rts.pools[primary.NodeID], val, numEvidence)
 
 	// Manually add half the evidence to the secondary which will mark them as
 	// pending.
 	for i := 0; i < numEvidence/2; i++ {
-		require.NoError(t, secondary.pool.AddEvidence(evList[i]))
+		require.NoError(t, rts.pools[secondary.NodeID].AddEvidence(evList[i]))
 	}
 
 	// the secondary should have half the evidence as pending
-	require.Equal(t, uint32(numEvidence/2), secondary.pool.Size())
+	require.Equal(t, numEvidence/2, int(rts.pools[secondary.NodeID].Size()))
 
 	state, err := stateDB2.Load()
 	require.NoError(t, err)
 
 	// update the secondary's pool such that all pending evidence is committed
 	state.LastBlockHeight++
-	secondary.pool.Update(state, evList[:numEvidence/2])
+	rts.pools[secondary.NodeID].Update(state, evList[:numEvidence/2])
 
 	// the secondary should have half the evidence as committed
-	require.Equal(t, uint32(0), secondary.pool.Size())
+	require.Equal(t, 0, int(rts.pools[secondary.NodeID].Size()))
 
-	// Simulate a router by listening for all outbound envelopes and proxying the
-	// envelopes to the respective peer (suite).
-	wg := new(sync.WaitGroup)
-	simulateRouter(wg, primary, testSuites, numEvidence)
-
-	// add the secondary reactor as a peer to the primary reactor
-	primary.peerUpdatesCh <- p2p.PeerUpdate{
-		Status: p2p.PeerStatusUp,
-		NodeID: secondary.peerID,
-	}
+	// start the network and ensure it's configured
+	rts.start(t)
+	// without the following sleep the test consistently fails.
+	time.Sleep(2 * time.Millisecond)
```

Comment:
I've updated the comment, but I suspect it's because the sleep forces a context switch in the thread scheduler that lets the router's threads do their thing.